xbmc/interfaces/legacy/Monitor.h

```diff
@@ -161,7 +161,7 @@ namespace XBMCAddon
        *\n
        * Will be called when XBMC receives or sends a notification\n
        */
-      virtual void    onNotification(const String sender, const String method, const String data) { XBMC_TRACE; }
+      virtual void    onNotification(const String& sender, const String& method, const String& data) { XBMC_TRACE; }
```

Comment:
I guess these changes yield a compile error. You should changes the implementations at line 42ff accordingly, imo.
Haven't you compile checked your changes?